Xamarin.Forms.Platform.iOS/Renderers/MediaElementRenderer.cs

```diff
@@ -0,0 +1,475 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using AVFoundation;
+using AVKit;
+using CoreGraphics;
+using CoreMedia;
+using Foundation;
+using UIKit;
+using Xamarin.Forms.Internals;
+
+namespace Xamarin.Forms.Platform.iOS
+{
+	public sealed class MediaElementRenderer : UIView, IVisualElementRenderer, IEffectControlProvider
+	{
+		MediaElement MediaElement { get; set; }
+		IMediaElementController Controller => MediaElement as IMediaElementController;
+
+#pragma warning disable 0414
+		VisualElementTracker _tracker;
+#pragma warning restore 0414
+
+		AVPlayerViewController _avPlayerViewController = new AVPlayerViewController();
+
+		NSObject _playToEndObserver;
+		NSObject _statusObserver;
+		NSObject _rateObserver;
+
+		public MediaElementRenderer()
+		{
+			_playToEndObserver = NSNotificationCenter.DefaultCenter.AddObserver(AVPlayerItem.DidPlayToEndTimeNotification, PlayedToEnd);
+			_avPlayerViewController.View.Frame = Bounds;
+			_avPlayerViewController.View.AutoresizingMask = UIViewAutoresizing.FlexibleWidth | UIViewAutoresizing.FlexibleHeight;
+			AddSubview(_avPlayerViewController.View);
+		}
+
+		VisualElement IVisualElementRenderer.Element => MediaElement;
+
+		UIView IVisualElementRenderer.NativeView => this;
+
+		UIViewController IVisualElementRenderer.ViewController => _avPlayerViewController;
+		
+		bool _idleTimerDisabled = false;
+		void SetKeepScreenOn(bool value)
+		{
+			if (value)
+			{
+				if (!UIApplication.SharedApplication.IdleTimerDisabled)
+				{
+					_idleTimerDisabled = true;
+					UIApplication.SharedApplication.IdleTimerDisabled = true;
+				}
+			}
+			else
+			{
+				if (_idleTimerDisabled)
+				{
+					_idleTimerDisabled = false;
+					UIApplication.SharedApplication.IdleTimerDisabled = false;
+				}
+			}
```

Comment:
_idleTimerDisabled keeps track of when this control requested IdleTimerDisabled and it was not already set. This means that if this value is false but the system-wide setting is true we won't disable it because it has been set elsewhere in the app.